packages/sdk/src/modules/did/index.ts

```diff
@@ -152,24 +162,44 @@ export class Did extends Base {
     return `did:peaq:${address}`;
   }
 
-  private _createVerificationMethod(address: Address) {
+  private _createVerificationMethod(verification: Verification, address: Address) {
     const id = uuidv4();
     const verificationMethod = new peaqDidProto.VerificationMethod();
 
+    // does this id change if it is Ed25519 vs Sr25519?
     verificationMethod.setId(id);
-    verificationMethod.setType(
-      peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020
-    );
+
+    if (verification.type !== peaqDidProto.VerificationType.ED25519VERIFICATIONKEY2020 &&
+      verification.type !== peaqDidProto.VerificationType.SR25519VERIFICATIONKEY2020) {
+      throw new Error('Invalid type: Type must be either 0: ED25519VERIFICATIONKEY2020 or 1: SR25519VERIFICATIONKEY2020');
+    }
+
+    verificationMethod.setType(verification.type);
     verificationMethod.setController(this._getDidId(address));
     verificationMethod.setPublickeymultibase(`z${address}`);
```

Comment:
How should the publicKeyMultibase be generated?

Reply:
publicKeyMultibase is the actual 64 bit public key of the did ss58 address which e.g `e02ebd2d5fcce127d4673c82d84bbff34fed7dc55f5fff6641acd0a1de343622`